src/parser.jl

```diff
@@ -2561,7 +2562,8 @@ end
 # flisp: parse-comprehension
 function parse_comprehension(ps::ParseState, mark, closer)
     ps = ParseState(ps, whitespace_newline=true,
-                    space_sensitive=false)
+                    space_sensitive=false,
+                    end_symbol=false)
```

Comment:
The thing that confuses me about this is why the flisp parser doesn't seem to have this state change within `parse-comprehension`. So I wonder whether this is the right fix, or whether we're still missing something here.

Reply:
Definitely possible. I haven't dug into the scheme parser much (because I find it borderline unreadable :P), but this seems correct to me regardless. Note that Julia currently can't handle this syntax in the general case:
```
julia> Meta.parse("""
       Any[foo(i) for i in x if begin
               true
           end
       ]
       """)
ERROR: Base.Meta.ParseError("expected \"]\"")
```
The newline before `for` is required for this to work.

Reply:
Yeah that's quite weird.  Certainly seems like a bug in `Meta.parse()`.

I'm keeping a list of such bugs in the README - can you add it here? https://github.com/JuliaLang/JuliaSyntax.jl#flisp-parser-bugs

Reply:
Done.